Add example for MS SQL Server

jbauerrfid opened the pull request:
# Description
Add example for MS SQL Server

# Reasons
Running MS SQL Server is a bit tricky in CircleCI. I want to share the build script worked out to save time for other devs.
Doing tests with MS SQL Server appears to be a common tasks, and I did not find anything functional in the CircleCI documentation.

devinmarieb commented:
Hi @jbauerrfid, thanks for taking the time to make this example project, but we unfortunately cannot include examples in our docs that rely on community orbs like cobli/wait-for@0.0.2 since we have no control over the maintenance or security of them.

jbauerrfid commented:
Hi @devinmarieb
I have updated the PR and replaced the ORB reference.
I would also propose to update your https://github.com/circleci/circleci-docs/blob/master/docs/CONTRIBUTING.md and add clarification about third party code like ORBs.

devinmarieb commented:
@jbauerrfid Thanks! Sorry for the delay. We've actually been having larger conversations around the best way to include projects like the one you originally proposed, where we can't include them in the official docs, but maybe we can have a new section for projects that use third party stuff. I appreciate the effort! We haven't forgotten about this :)

jbauerrfid commented:
Hi @devinmarieb 

CircleCI is pretty quick at responding to contributions. I have open request on other OSS projects, where I wait for months or get no response from the maintainers at all. 

So when you want to reorganize third party contributions it is ok for me.

devinmarieb commented:
@jbauerrfid Thanks! I'm going to keep this PR open until we make a final decision on what to do with projects that use third party tools. These example projects people suggest are very useful for others in similar situations, but we just don't have a good way of including them in our current structure. It's possible you will see it merged or a request for changes in a few months time :) Thanks for your suggestions!

jbauerrfid commented:
Hi @devinmarieb
did you see that I replaced the ORB code in my PR?

devinmarieb commented:
@jbauerrfid yes I did :) There's actually further internal discussion on this whole page in general now which is why I haven't merged yet.

felicianotech commented:
We need to re-review this. @circleci/dre
